### PR TITLE
colwise functions can use local bvariables and columns in lambdas.

### PR DESCRIPTION
**Pull request by romainfrancois**
``` r
library(dplyr, warn.conflicts = FALSE)

local({
  value <- 10
  iris %>% 
    summarise_at("Sepal.Length", ~ sum(. / value))
})
#>   Sepal.Length
#> 1        87.65

local({
  iris %>% 
    summarise_at("Sepal.Length", ~ sum(. / Sepal.Width))
})
#>   Sepal.Length
#> 1     293.0521
```

<sup>Created on 2019-04-05 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

### CODE REVIEW

File: inst/include/tools/Quosure.h [hunk 1]
```diff
@@ -61,11 +61,23 @@ class NamedQuosure {
...
+  // making a new quosure so that the quosure of the lambda is unchanged, but the
+  // expression is data masked via eval_tidy(),
+  // i.e. going from
+  //
+  // (function(.) some_expression(., <column>, <local object>))(Sepal.Length)
+  // to
+  // (function(.) rlang::eval_tidy(quote(some_expression(., <column>, <local object>)(Sepal.Length)), <data mask>)
+  Rcpp::Shelter<SEXP> local;
+
+  SEXP expr = local(Rf_duplicate(named_quosure.expr()));
```
Comment (lionel-) [inline inst/include/tools/Quosure.h]: Duplication should be shallow, especially with quasiquotation. Unquoted objects may be heavy.

**Comment by lionel-** [comment]
Je vais essayer de simplifier encore un peu.

**Comment by lionel-** [comment]
How does this look?

File: src/init.cpp [hunk 2]
```diff
@@ -97,7 +97,7 @@ SEXP symbols::indices = Rf_install("indices");
 SEXP symbols::ptype = Rf_install("ptype");
 SEXP symbols::names = R_NamesSymbol;
 
-SEXP symbols::rlang_eval_bare = mark_precious(Rf_lang3(R_DoubleColonSymbol,
+SEXP symbols::rlang_eval_bare = mark_precious(Rf_lang3(symbols::double_colon,
```
Comment (romainfrancois) [inline src/init.cpp]: Thanks. I prefer using `symbols::` anyway, but 3.1 support is getting dropped in #4320

**Comment by romainfrancois** [comment]
Thanks @lionel-. I added some more to short-circuit this when hybrid evaluation can happen.

**Pull request merged**